lib/DirectEditing.js

```diff
@@ -87,19 +87,23 @@ DirectEditing.prototype.complete = function() {
     return;
   }
 
-  var containerBounds, bounds;
-
-  var text = this.getValue();
-
-  if (text !== active.context.text || this.resizable) {
-    containerBounds = this._textbox.container.getBoundingClientRect();
-    bounds = this.$textbox.getBoundingClientRect();
-
-    active.provider.update(active.element, text, active.context.text, {
-      x: bounds.left - containerBounds.left,
-      y: bounds.top - containerBounds.top,
-      width: bounds.width,
-      height: bounds.height
+  var previousBounds = active.context.bounds,
+      newBounds = this.$textbox.getBoundingClientRect(),
+      newText = this.getValue(),
+      previousText = active.context.text;
+
+  if (
+    newText !== previousText ||
+    newBounds.height !== previousBounds.height ||
+    newBounds.width !== previousBounds.width
+  ) {
+    var containerBounds = this._textbox.container.getBoundingClientRect();
```

Comment:
This variable assignment will be lifted up to the outer scope. Please declare it in the method root.